lm_eval/models/optimum_lm_genai.py

```diff
@@ -0,0 +1,164 @@
+from importlib.util import find_spec
+from pathlib import Path
+
+from lm_eval.api.registry import register_model
+from lm_eval.models.huggingface import HFLM
+
+import sys
+for path in sys.path:
+    print(path)
+
+from llm_bench.python.llm_bench_utils.ov_utils import create_text_gen_model, build_ov_tokenizer
+from llm_bench.python.llm_bench_utils.model_utils import get_use_case
+
+
+@register_model("optimum-causal")
+class OptimumLM_2(HFLM):
+    """
+    Optimum Intel provides a simple interface to optimize Transformer models and convert them to \
+    OpenVINO™ Intermediate Representation (IR) format to accelerate end-to-end pipelines on \
+    Intel® architectures using OpenVINO™ runtime.
+    """
+
+    def __init__(
+        self,
+        device="cpu",
+        convert_tokenizer = False,
+        trust_remote_code = True,
+        kv_cache = False,
+        **kwargs,
+    ) -> None:
+        # if "backend" in kwargs:
+        #     # optimum currently only supports causal models
+        #     assert (
+        #         kwargs["backend"] == "causal" 
+        #     ), "Currently, only OVModelForCausalLM is supported."
+
+        self.openvino_device = device
+        self.trust_remote_code = trust_remote_code,
+        self.convert_tokenizer = convert_tokenizer
+        self.kv_cache = kv_cache
+
+        super().__init__(
+            device=self.openvino_device,
+            backend=kwargs.pop("backend", "causal"),
+            **kwargs,
+        )
+
+    def _create_model(
+        self,
+        pretrained: str,
+        revision="main",
+        dtype="auto",
+        trust_remote_code=True,
+        **kwargs,
+    ) -> None:
+        DEFAULT_LLM_BENCH_ARGS = {
+            'model': '',
+            'model_id': '',
+            'framework': 'ov',
+            'infer_count': None,
+            'num_iters': 0,
+            'images': None,
+            'seed': 42,
+            'load_config': None,
+            'memory_consumption': 0,
+            'batch_size': 1,
+            'fuse_decoding_strategy': False,
+            'make_stateful': False,
+            'save_prepared_model': None,
+            'num_beams': 1,
+            'fuse_cache_reorder': False,
+            'torch_compile_backend': 'openvino',
+        }
+
+        MODEL_TYPE_BY_MODEL_NAME = {
```

Comment:
@sstrehlk, I dont' this it is a good idea to introduce this in the lm_eval. Can we keep this in Optimum, let's say, and just import here?

Reply:
MODEL_TYPE_BY_MODEL_NAME dictionary was removed from file.